capsules/src/ble_advertising_driver.rs

```diff
@@ -836,19 +548,27 @@ where
         &self,
         command_num: usize,
         data: usize,
-        _: usize,
+        interval: usize,
         appid: kernel::AppId,
     ) -> ReturnCode {
         match command_num {
-            // Start periodic advertisments
+            // Start periodic advertisements
             0 => self.app
                 .enter(appid, |app, _| {
                     if let Some(BLEState::Initialized) = app.process_status {
-                        app.process_status = Some(BLEState::AdvertisingIdle);
-                        app.random_nonce = self.alarm.now();
-                        app.set_next_alarm::<A::Frequency>(self.alarm.now());
-                        self.reset_active_alarm();
-                        ReturnCode::SUCCESS
+                        let pdu_type = data as AdvPduType;
+                        match pdu_type {
+                            ADV_IND | ADV_NONCONN_IND | ADV_SCAN_IND => {
```

Comment:
Shouldn't it be an error to be ADV_IND or ADV_SCAN_IND modes since we don't support scan requests/responses or connections?

Reply:
My opinion is that enforcing behavior is the correct approach. If we're just providing a 2.4 GHz radio interface, then users should be able to do whatever. But if we're providing a BLE interface, then actually enforcing BLE behavior makes it easier for users to do things correctly. I agree that we are also concerned with whether apps influence each other, but we also want to support even a single application attempting to function correctly.

Reply:
OK, so high level, this driver should provide an interface to the the link-layer as specified in Vol 6. Part B. of the BLE specification. Meaning, if you send a scannable advertisement the driver should respond to scans, but you can send whatever ADV DATA you want.

That's exactly what this does, _except_ it doesn't yet respond to scan requests, but that will be merged soon.

Reply:
Yeah, I agree with that.

And it isn't the biggest problem in the world to be using ADV_IND or ADV_SCAN_IND without scan responses or connections. It's not like it breaks BLE or anything. We can keep them if you think that's the right choice. Generally though, I would suggest it be an error to try to use modes we don't support, for instance directed advertisements should be an error so users don't mistakenly think we support them.